discovery/syncer.go

```diff
@@ -970,6 +1020,113 @@ func (g *GossipSyncer) ProcessQueryMsg(msg lnwire.Message, peerQuit <-chan struc
 	}
 }
 
+// ProcessSyncTransition sends a request to the gossip syncer to transition its
+// sync type to a new one.
+func (g *GossipSyncer) ProcessSyncTransition(newSyncType SyncerType) error {
+	errChan := make(chan error, 1)
+	select {
+	case g.syncTransitions <- &syncTransitionReq{
+		newSyncType: newSyncType,
+		errChan:     errChan,
+	}:
+	case <-time.After(syncTransitionTimeout):
+		return ErrSyncTransitionTimeout
+	case <-g.quit:
+		return ErrGossipSyncerExiting
+	}
+
+	select {
+	case err := <-errChan:
+		return err
+	case <-g.quit:
+		return ErrGossipSyncerExiting
+	}
+}
+
+// handleSyncTransition handles a new sync type transition request.
+//
+// NOTE: The gossip syncer might have another sync state as a result of this
+// transition.
+func (g *GossipSyncer) handleSyncTransition(req *syncTransitionReq) error {
+	syncType := SyncerType(atomic.LoadUint32(&g.syncType))
+	log.Debugf("GossipSyncer(%x): transitioning from %v to %v",
+		g.cfg.peerPub, syncType, req.newSyncType)
+
+	switch req.newSyncType {
+	// If an active sync has been requested, then we should resume receiving
+	// new graph updates from the remote peer.
+	case ActiveSync:
+		// We'll avoid sending a timestamp range if we already have an
+		// ActiveSync type.
+		if syncType == ActiveSync {
+			return nil
+		}
+
+		err := g.sendGossipTimestampRange(time.Now(), math.MaxUint32)
+		if err != nil {
+			return fmt.Errorf("unable to send active update "+
+				"horizon: %v", err)
+		}
+
+		// If the gossip syncer currently has a PassiveSync type, then
+		// we'll revert to our initial syncingChans state to ensure we
+		// can reconcile any channels we're missing that the remote peer
+		// has. This isn't needed when transitioning from a HistoricalSync, as
+		// both sides have synced their complete graphs with each other.
+		if syncType == PassiveSync {
+			atomic.StoreUint32(&g.state, uint32(syncingChans))
```

Comment:
Not really other than some minimal bandwidth gains. Since `PassiveSync` is the only type that doesn't require sending out our set of channels, we make sure to do so when transitioning to another type.